docs/design/runtime_processing/runtime_recording.md

```diff
@@ -0,0 +1,387 @@
+
+@15
+User1 is "Idle"
+Hook is WAIT
+User1 -> Hook : End recording \n[ /caret/end_record ]\n\n
+User1 -> Lttng
+Lttng is "Destroyed"
+```
+
+A detailed sequence diagram is written in [Sequence](#sequence).
+
+Trace nodes uses topic communication for control and notification.
+
+In addition, to maintain compatibility of usage,
+in cases where sessions are started in advance, they can record correctly.
+
+```plantuml
+concise "Terminal 0" as User0
+concise "trace node state" as Hook
+concise "Terminal 1" as User2
+concise "LTTng" as Lttng
+
+@0
+User0 is ""
+User2 is ""
+Hook is ""
+Lttng is "idle"
+
+@3
+User2 is "ros2 caret trace"
+User2 -> Lttng : Start session\n
+Lttng is "Active"
+
+@5
+User0 is "launch"
+Hook is RECORD : Record initialization and runtime tracepoints
+User0 -> Hook
+
+
+@15
+User2 is ""
+User2 -> Lttng : Destroy session\n
+Lttng is "Destroyed"
+
+```
+
+Note that Initialization trace data is recorded for each LTTng session.
+
+The main state transitions are shown below.
+
+```plantuml
+[*] --> RECORD: An active lttng session exists
+[*] --> WAIT: No active lttng session exists
+WAIT : Stores initialization trace data in memory
+PREPARE : Record initialization trace data with LTTng
+WAIT --> PREPARE :Start recording
+
+RECORD --> WAIT : End recording
+PREPARE --> RECORD : Finished recording stored initialization trace data
+RECORD : Record initialization and runtime trace data with LTTng
+```
+
+Detailed state transitions are written in [Status](#status).
+
+## Topic
+
+Runtime recording uses the following topic communication.
+
+| topic name            | message type | role                                          |
+| --------------------- | ------------ | --------------------------------------------- |
+| `/caret/start_record` | Start.msg    | Start recording. Transition to PREPARE state. |
+| `/caret/end_record`   | End.msg      | End recording. Transition to WAIT state.      |
+| `/caret/status`       | Status.msg   | Sync current recording state.                 |
+
+### Start.msg
+
+```cpp
+uint32 recording_frequency 100
+string ignore_nodes  # reserved
+string ignore_topics # reserved
+string select_nodes  # reserved
+string select_topics # reserved
+```
+
+The recording_frequency is the frequency at which each process records initialization trace data in the PREPARE state.
+The higher the frequency, the shorter the time required to complete the PREPARE state, but the greater the risk of tracer discarded.
+
+`ignore_nodes` `ignore_topics` `select_nodes` `select_topics` are unimplemented features.
+They are reserved fields for setting [tracepoint filtering](./tracepoint_filtering.md) at the start of the measurement from CLI.
+
+<prettier-ignore-start>
+!!!Info
+    As another method of recording_frequency,
+    initialization-related trace points can be recorded in [blocking mode](https://lttng.org/blog/2017/11/22/lttng-ust-blocking-mode/) to prevent tracer discarded.
```

Comment:
```suggestion
    [blocking mode](https://lttng.org/blog/2017/11/22/lttng-ust-blocking-mode/) might be a reasonable choice to avoid tracer discarded during storing meta-data into a disk.
```

Reply:
fixed at https://github.com/hsgwa/CARET_doc/pull/5